addons/survey/models/survey_question.py

```diff
@@ -651,6 +651,23 @@ def _get_stats_summary_data_scored(self, user_input_lines):
             'right_inputs_count': len(user_input_lines.filtered(lambda line: line.answer_is_correct).mapped('user_input_id'))
         }
 
+    # ------------------------------------------------------------
+    # OTHERS
+    # ------------------------------------------------------------
+
+    def _get_correct_answer(self):
+        """ Return the question correct answer if it has one, otherwise return False. """
+        self.ensure_one()
+        correct_answer = False
+        if self.question_type in ['numerical_box', 'date', 'datetime']:
+            correct_answer = self[f'answer_{self.question_type}']
+            if self.question_type == 'date':
+                correct_answer = tools.format_date(self.env, correct_answer)
+            elif self.question_type == 'datetime':
+                correct_answer = tools.format_datetime(self.env, correct_answer, tz='UTC', dt_format=False)
+        elif self.question_type in ['simple_choice', 'multiple_choice']:
+            correct_answer = self.suggested_answer_ids.filtered('is_correct').ids
```

Comment:
This line will trigger a query. If this method is not called in batch for a set of questions, it can be substancial.
I would recommend allowing a multiple records recordset for this, and return a dictionary. 
Sorry it just appeared to me.